src/static/helpers/passwordlessHelper.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ShopperLogin, TokenResponse} from '../../lib/shopperLogin';
+import {createCodeVerifier, stringToBase64} from './slasHelper';
+
+/**
+ * Function to send passwordless login token
+ * **Note** At the moment, passwordless is only supported on private client
+ * @param slasClient a configured instance of the ShopperLogin SDK client.
+ * @param credentials - the id and password and clientSecret (if applicable) to login with.
+ * @param credentials.clientSecret? - secret associated with client ID
+ * @param parameters - parameters to pass in the API calls.
+ * @param parameters.callbackURI? - URI to send the passwordless login token to. Must be listed in your SLAS configuration. Required when mode is callback
+ * @param parameters.usid? - Unique Shopper Identifier to enable personalization.
+ * @param parameters.userid - User Id for login
+ * @param parameters.locale - The locale of the template. Not needed for the callback mode
+ * @param parameters.mode - Medium of sending login token
+ * @returns Promise of Response
+ */
+export async function authorizePasswordless(
+  slasClient: ShopperLogin<{
+    shortCode: string;
+    organizationId: string;
+    clientId: string;
+    siteId: string;
+  }>,
+  credentials: {
+    clientSecret: string;
+  },
+  parameters: {
+    callbackURI?: string;
+    usid?: string;
+    userid: string;
+    locale?: string;
+    mode: string;
+  }
+): Promise<Response> {
+  const authHeaderIdSecret = `Basic ${stringToBase64(
+    `${slasClient.clientConfig.parameters.clientId}:${credentials.clientSecret}`
+  )}`;
+  const tokenBody = {
+    user_id: parameters.userid,
+    mode: parameters.mode,
```

Comment:
NIT: It doesn't look like we have have a pattern for throwing when required params are not included, so I don't expect you to add it now because I like consistency more. But I'll take note on that.

Reply:
It also seems like typescript checks for the existence of these params that are directly required. But when it's parameters within an object like ShopperLogin, typescript doesn't automatically give an error that expected arguments don't exist

Reply:
We shouldn't reply on TypeScript types for things like that because this lib is compiled into JS and it can be used by a project that isn't TS. Typescript won't bake in any kind of parameter checking so there is the possibility that you this code would make a server request that would for sure fail if that param is required. 

But again, no need to change that now, it doesn't seem like it's a pattern that we have, and we should probably fix that in another PR in one big swoop.